### PR TITLE
replace groupN actions with granular subcommands

### PR DESCRIPTION
**Pull request by A-childs-encyclopedia**
cc @badosu @Beherith

### CODE REVIEW

**Comment by A-childs-encyclopedia** [comment]
This pull request replaces groupN (replace n with any number from 0 to 9) actions in favour of the new granular subcommands that were added. It also improves luaui/Widgets/cmd_unitgroup_clear_selection_on_empty.lua by incorporating changes from @badosu

**Comment by A-childs-encyclopedia** [comment]
link to adjustment by badosu :- [link](https://gist.github.com/badosu/4f669cd06966430839102036d10fba16)

**Comment by badosu** [comment]
@A-childs-encyclopedia looks great! However the place you're replacing is the list of bindings we _unbind_ from engine.

The place we need to change is here: https://github.com/beyond-all-reason/Beyond-All-Reason/blob/64646c05a41ecccf6b32c953e74d5f38a7f69ea2/luaui/Widgets/cmd_bar_hotkeys.lua#L354-L364

**Comment by A-childs-encyclopedia** [comment]
I updated the file with the changes 👍🏼

File: luaui/Widgets/cmd_bar_hotkeys.lua [hunk 1]
```diff
@@ -312,16 +312,56 @@ local function makeBindsTable(keyLayout)
...
-		{ "Any+2", "group2" },
-		{ "Any+3", "group3" },
-		{ "Any+4", "group4" },
-		{ "Any+5", "group5" },
-		{ "Any+6", "group6" },
-		{ "Any+7", "group7" },
-		{ "Any+8", "group8" },
-		{ "Any+9", "group9" },
+		{            "0", "group", "0"            },
+{       "Ctrl0+0", "group", "set 0"        },
```
Comment (badosu) [inline luaui/Widgets/cmd_bar_hotkeys.lua:316]: This typo is still present.

**Comment by A-childs-encyclopedia** [comment]
fixed typo

File: luaui/Widgets/cmd_bar_hotkeys.lua [hunk 2]
```diff
@@ -312,16 +312,56 @@ local function makeBindsTable(keyLayout)
...
-		{ "Any+1", "group1" },
-		{ "Any+2", "group2" },
-		{ "Any+3", "group3" },
-		{ "Any+4", "group4" },
-		{ "Any+5", "group5" },
-		{ "Any+6", "group6" },
-		{ "Any+7", "group7" },
-		{ "Any+8", "group8" },
-		{ "Any+9", "group9" },
+{            "0", "group", "0"            },
```
Comment (badosu) [inline luaui/Widgets/cmd_bar_hotkeys.lua:315]: The action names are a bit different from the new ones, I made the list to replace with here: https://gist.github.com/badosu/d7b4eae890f4b2888a7cb35ecc454dc1

Use this list and we're good to merge!

**Comment by A-childs-encyclopedia** [comment]
done

**Comment by badosu** [comment]
Thank you! cc @Beherith LGTM :+1:

File: luaui/Widgets/cmd_bar_hotkeys.lua [hunk 3]
```diff
@@ -311,17 +311,57 @@ local function makeBindsTable(keyLayout)
...
-		{ "Any+2", "group2" },
-		{ "Any+3", "group3" },
-		{ "Any+4", "group4" },
-		{ "Any+5", "group5" },
-		{ "Any+6", "group6" },
-		{ "Any+7", "group7" },
-		{ "Any+8", "group8" },
-		{ "Any+9", "group9" },
+		
+{            "0", "group", "0"            },
```
Comment (badosu) [inline luaui/Widgets/cmd_bar_hotkeys.lua:315]: You can remove these binds and change on 622:

```
    for i = 0, 9 do
        table.insert(_binds, { 'Alt+'..i , "add_to_autogroup", i })

        table.insert(_binds, { i                , "group", i                })
        table.insert(_binds, { 'Ctrl+'..i       , "group", "set "..i        })
        table.insert(_binds, { 'Shift+'..i      , "group", "selectadd "..i })
        table.insert(_binds, { 'Ctrl+Shift+'..i , "group", "add "..i        })
        table.insert(_binds, { 'Alt+'..i        , "group", "selecttoggle "..i     })
    end
```

**Comment by A-childs-encyclopedia** [comment]
Like this? (The latest commit)

**Comment by A-childs-encyclopedia** [comment]
Wait no I made the wrong change lol

File: luaui/Widgets/cmd_bar_hotkeys.lua [hunk 4]
```diff
@@ -311,17 +311,57 @@ local function makeBindsTable(keyLayout)
...
+{       "Ctrl+0", "group", "set 0"        },
```
Comment (badosu) [inline luaui/Widgets/cmd_bar_hotkeys.lua:316]: Now you can remove these binds

**Comment by A-childs-encyclopedia** [comment]
Maybe I did make the right change. I'm getting confused

File: luaui/Widgets/cmd_bar_hotkeys.lua (hunk 4 above)
Comment (A-childs-encyclopedia) [reply luaui/Widgets/cmd_bar_hotkeys.lua:316]: I'll do it tomorrow morning

Comment (badosu) [reply luaui/Widgets/cmd_bar_hotkeys.lua:316]: Ok, @Beherith dont merge before this

**Comment by badosu** [comment]
It's alright, just need to remove these lines: https://github.com/beyond-all-reason/Beyond-All-Reason/pull/982/files#diff-397d0d94a127186f670fe26a12a73d3d0685de10d75184621a220f6e047e8ca4R314-R364

And you're good to go.

The reason is that now we add the binds on the place where we define the `add_to_autogroup` binds, so we don't need them defined explicitly there.

**Comment by A-childs-encyclopedia** [comment]
@badosu that change is done too 👍🏼

**Comment by badosu** [comment]
LGTM :+1: Good job!

**Comment by A-childs-encyclopedia** [comment]
thx

**Pull request merged**